File: LeetCode/Easy/0492-construct-the-rectangle/0492-construct-the-rectangle.java
class Solution {
    public int[] constructRectangle(int area) {
        int[] ans = new int[2];
        int sqrt = (int)Math.sqrt(area);
        int max = 1;


        for (int i = 2; i <= sqrt; i++) {
            if(area % i == 0) {
                max = i;
            }
        }

        ans[0] = area / max;
        ans[1] = max;

        return ans;
    }
}

File: LeetCode/Easy/0492-construct-the-rectangle/0492-construct-the-rectangle.kt
class Solution {
    fun constructRectangle(area: Int): IntArray {
        val sqrt = Math.sqrt(area.toDouble()).toInt()
        var max: Int = 1

        for(i in 2..sqrt) {
            if (area % i == 0) max = i
        }

        return intArrayOf(area / max, max)
    }
}